pkg/options/start.go

```diff
@@ -54,6 +63,33 @@ func resourceByType(resources pipelineResources, restype string) []string {
 	return []string{}
 }
 
+func (taskRunOpts *TaskRunOpts) UseTaskRunFrom(tr *v1beta1.TaskRun, cs *cli.Clients, tname string, taskKind string) error {
+	var (
+		trUsed *v1beta1.TaskRun
+		err    error
+	)
+	if taskRunOpts.Last {
+		trUsed, err = task.LastRun(cs, tname, taskRunOpts.CliParams.Namespace(), taskKind)
+		if err != nil {
+			return err
+		}
+	} else if taskRunOpts.UseTaskRun != "" {
+		trUsed, err = tractions.Get(cs, taskRunOpts.UseTaskRun, metav1.GetOptions{}, taskRunOpts.CliParams.Namespace())
+		if err != nil {
+			return err
+		}
+	}
+
```

Comment:
There is one last piece of validation needed here. We need to make sure the TaskRun from `--last` or `--use-taskrun` was originally created by the Task or ClusterTask being used in a start command.

As of now, this will allow users to start TaskRuns using any previous TaskRun regardless of whether it is associated with the Task or ClusterTask used in a start command.

We should verify that the TaskRef name of the previous TaskRun matches the TaskRef name of the TaskRun that is about to be created. Also, we should verify if `taskKind` is `ClusterTask` whether the TaskRef kind is ClusterTask, which [we do here for clustertask start](https://github.com/tektoncd/cli/blob/53e4a0ca735ddf4ed64bf3d226953c43854f4da8/pkg/cmd/clustertask/start.go#L200).

Reply:
For `--use-taskrun` validation was not there but with `--last` TaskRuns were being filtered according to kind